src/sentry/deletions/defaults/monitor.py

```diff
@@ -12,8 +7,6 @@ def get_child_relations(self, instance: Monitor) -> list[BaseRelation]:
         from sentry.monitors import models
 
         return [
-            ModelRelation(
-                models.MonitorCheckIn, {"monitor_id": instance.id}, BulkModelDeletionTask
-            ),
+            ModelRelation(models.MonitorCheckIn, {"monitor_id": instance.id}),
```

Comment:
I think we need to add in `MonitorIncident` here, before `MonitorCheckin`. Then all the `MonitorIncident` rows will be removed first, and we can use the bulk delete on `MonitorCheckin` still.